src/factories/request.ts

```diff
@@ -47,6 +51,26 @@ export default class Request {
     );
   };
 
+  // Step 1: Create a new method `setAxiosConfig`
+  setAxiosConfig = (customConfig: AxiosRequestConfig) => {
```

Comment:
this method is no longer needed with this approach.

Reply:
5ed40c508aa410cc3eeac06c35c265dadfd4928c